tools/indexer/main.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2018 Microsoft Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+type packageSet map[string]bool
+
+// adds any missing SDK packages to godoc.org
+func main() {
+	dir := ""
+	if len(os.Args) > 1 {
+		// assume second arg is source dir
+		dir = os.Args[1]
+	} else {
+		// if no args specified assume we're running from the source dir
+		// and calculate the relative path to the services directory.
+		var err error
+		dir, err = filepath.Abs("../../services")
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	pkgs, err := getPackagesForIndexing(dir)
+	if err != nil {
+		panic(err)
+	}
+
+	indexedPkgs, err := getIndexedPackages()
+	if err != nil {
+		panic(err)
+	}
+
+	// for each package in pkgs, check if it's already been
+	// indexed.  if it hasn't been indexed then do so
+
+	for pkg := range pkgs {
+		if _, already := indexedPkgs[pkg]; already {
+			pkgs[pkg] = true
+			continue
+		}
+
+		fmt.Printf("indexing %s...", pkg)
+		resp, err := http.DefaultClient.Get(fmt.Sprintf("https://godoc.org/%s", pkg))
+		if err != nil {
+			panic(err)
+		}
+
+		resp.Body.Close()
+		if resp.StatusCode == http.StatusOK || resp.StatusCode == http.StatusNotFound {
+			// a 404 means that the package exists locally but not yet in github
+			fmt.Printf("completed with status '%s'\n", resp.Status)
+			pkgs[pkg] = true
+		} else {
+			fmt.Printf("FAILED with status '%s'\n", resp.Status)
+		}
+
+		// sleep a bit between indexing
+		time.Sleep(60 * time.Second)
```

Comment:
>time.Sleep(60 * time.Second) [](start = 2, length = 28)

is this to avoid getting blocked by some kind of bot protection ? Maybe we can do a random number between 30-70 or something if you saw that there is still an issue :D

Reply:
You know this might not be an issue any more, I will bump the sleep to 10 seconds and if we have problems in the future we can adjust.